### PR TITLE
Move Sending feature logic to TransportComponent

### PR DESCRIPTION
**Pull request by timbussmann**

### CODE REVIEW

File: src/NServiceBus.Core/StartableEndpoint.cs [hunk 1]
```diff
@@ -36,13 +36,18 @@ public async Task<IEndpointInstance> Start()
...
 
             var messageOperations = sendComponent.CreateMessageOperations(builder, pipelineComponent);
             var messageSession = new MessageSession(pipelineComponent.CreateRootContext(builder, messageOperations));
 
             AppDomain.CurrentDomain.SetPrincipalPolicy(PrincipalPolicy.WindowsPrincipal);
 
             await receiveComponent.PrepareToStart(builder, recoverabilityComponent, messageOperations).ConfigureAwait(false);
+            
+            // This is a hack to maintain the current order of transport infrastructure initialization
+            await transportComponent.SendPreStartupChecks().ConfigureAwait(false);
```
Comment (andreasohlund) [inline src/NServiceBus.Core/StartableEndpoint.cs]: Rename to `InvokeSendPreStartupChecksForBackwardsCompatibility`?

File: src/NServiceBus.Core/StartableEndpoint.cs [hunk 2]
```diff
@@ -36,13 +36,18 @@ public async Task<IEndpointInstance> Start()
             await pipelineComponent.Start(builder).ConfigureAwait(false);
 
             await transportComponent.Start().ConfigureAwait(false);
+            // This is a hack to maintain the current order of transport infrastructure initialization
+            transportComponent.ConfigureSendInfrastructure();
```
Comment (andreasohlund) [inline src/NServiceBus.Core/StartableEndpoint.cs]: Rename to `ConfigureSendInfrastructureForBackwardsCompatibility`?

File: src/NServiceBus.Core/Transports/TransportComponent.cs [hunk 3]
```diff
@@ -67,13 +98,6 @@ public Task CreateQueuesIfNecessary(string username)
             return queueCreator.CreateQueueIfNecessary(QueueBindings, username);
         }
 
-        public QueueBindings QueueBindings { get; }
-
-        public IPushMessages BuildMessagePump()
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/TransportComponent.cs]: How where you able to remove this? (How do receive component get access to the message pump?)

Comment (timbussmann) [reply src/NServiceBus.Core/Transports/TransportComponent.cs]: there is a GetMessagePump factory method used by the receive component a bit further up 🤷‍♂

**Comment by timbussmann** [comment]
rebased

**Pull request merged**